bgpd/bgpd.c

```diff
@@ -202,6 +202,80 @@ int bgp_option_check(int flag)
 	return CHECK_FLAG(bm->options, flag);
 }
 
+/* set the bgp no-rib option during runtime and remove installed routes */
+int bgp_option_norib_set_runtime(void)
+{
+	struct bgp *bgp;
+	struct listnode *node;
+
+	/* check if option is already set in BM */
+	if (bgp_option_check(BGP_OPT_NO_FIB))
+		return 0;
+
+	/* set the no-rib option in BM */
+	bgp_option_set(BGP_OPT_NO_FIB);
+
+	/* informational log message */
+	zlog_info("Disabled BGP route installation to RIB (Zebra)");
+
+	/* iterate through bgp instances */
+	for (ALL_LIST_ELEMENTS_RO(bm->bgp, node, bgp)) {
+		if (!bgp)
+			continue;
+
+		/* Iterate through all RIB tables of the bgp instance and
+		   withdraw their routes from Zebra. */
+		for (afi_t afi = AFI_IP; afi < AFI_MAX; ++afi) {
+			for (safi_t safi = SAFI_UNICAST; safi < SAFI_MAX;
+			     ++safi) {
+				bgp_zebra_withdraw_table(bgp, afi, safi);
+			}
+		}
+	}
+
+	/* informational log message */
+	zlog_info("All routes have been withdrawn from RIB (Zebra)");
+
+	return 0;
+}
+
+/* unset the bgp no-rib option during runtime and announce routes to Zebra */
+int bgp_option_norib_unset_runtime(void)
+{
+	struct bgp *bgp;
+	struct listnode *node;
+
+	/* check if option is already unset in BM */
+	if (!bgp_option_check(BGP_OPT_NO_FIB))
+		return 0;
+
+	/* unset the no-rib option in BM */
+	bgp_option_unset(BGP_OPT_NO_FIB);
+
+	/* informational log message */
+	zlog_info("Enabled BGP route installation to RIB (Zebra)");
+
+	/* iterate through bgp instances */
+	for (ALL_LIST_ELEMENTS_RO(bm->bgp, node, bgp)) {
+		if (!bgp)
```

Comment:
As before how can bgp be NULL?

Reply:
As stated before, i just wanted to make sure i was not producing segfaults. Do you want me to remove it anyway?